src/raft/fixture.c

```diff
@@ -597,48 +698,23 @@ static int ioMethodAppend(struct raft_io *raft_io,
 	return 0;
 }
 
-static int ioMethodTruncate(struct raft_io *raft_io, raft_index index)
+static int ioMethodTruncate(struct raft_io *raft_io,
+			    struct raft_io_truncate *trunc,
+			    raft_index index)
 {
 	struct io *io = raft_io->impl;
-	size_t n;
-
-	n = (size_t)(index - 1); /* Number of entries left after truncation */
-
-	if (n > 0) {
-		struct raft_entry *entries;
-
-		/* Create a new array of entries holding the non-truncated
-		 * entries */
-		entries = raft_malloc(n * sizeof *entries);
-		if (entries == NULL) {
-			return RAFT_NOMEM;
-		}
-		memcpy(entries, io->entries, n * sizeof *io->entries);
-
-		/* Release any truncated entry */
-		if (io->entries != NULL) {
-			size_t i;
-			for (i = n; i < io->n; i++) {
-				raft_free(io->entries[i].buf.base);
-			}
-			raft_free(io->entries);
-		}
-		io->entries = entries;
-	} else {
-		/* Release everything we have */
-		if (io->entries != NULL) {
-			size_t i;
-			for (i = 0; i < io->n; i++) {
-				raft_free(io->entries[i].buf.base);
-			}
-			raft_free(io->entries);
-			io->entries = NULL;
-		}
-	}
-
-	io->n = n;
+	struct truncate *r;
 
+	sm_init(&trunc->sm, fio_invariant, NULL, fio_states, "fio-trunc", FIO_START);
+	r = raft_malloc(sizeof(*r));
+	assert(r != NULL);
+	r->type = TRUNCATE;
```

Comment:
It's not always clear how the structure with macro inside is initialized.
```
struct truncate {
	REQUEST;
	struct raft_io_truncate *req;
	raft_index index;
};
```
having initializer `*r = {type=TRUNCATE ... }` makes the reader not to dig into details assuming that non mentioned struct fields will be filled with zeros.